File: backend/app/schemas/medical_record.py
"""
Schemas para Medical Records (Historias Clínicas)
"""
from datetime import datetime
from typing import Optional
from pydantic import BaseModel


# ============= BASE SCHEMAS =============

class MedicalRecordBase(BaseModel):
    """Schema base para medical records"""
    patient_id: int
    appointment_id: Optional[int] = None
    consultation_date: datetime
    chief_complaint: Optional[str] = None
    symptoms: Optional[str] = None
    diagnosis: str
    treatment: Optional[str] = None
    notes: Optional[str] = None
    # Signos vitales
    blood_pressure: Optional[str] = None
    heart_rate: Optional[int] = None
    temperature: Optional[float] = None
    weight: Optional[float] = None
    height: Optional[float] = None


# ============= REQUEST SCHEMAS =============

class MedicalRecordCreate(MedicalRecordBase):
    """Schema para crear una historia clínica"""
    pass


class MedicalRecordUpdate(BaseModel):
    """Schema para actualizar una historia clínica"""
    patient_id: Optional[int] = None
    appointment_id: Optional[int] = None
    consultation_date: Optional[datetime] = None
    chief_complaint: Optional[str] = None
    symptoms: Optional[str] = None
    diagnosis: Optional[str] = None
    treatment: Optional[str] = None
    notes: Optional[str] = None
    blood_pressure: Optional[str] = None
    heart_rate: Optional[int] = None
    temperature: Optional[float] = None
    weight: Optional[float] = None
    height: Optional[float] = None


# ============= RESPONSE SCHEMAS =============

class MedicalRecordResponse(MedicalRecordBase):
    """Schema de respuesta con datos completos"""
    id: int
    consultorio_id: int
    doctor_id: int
    created_at: datetime
    updated_at: datetime
    
    # Datos relacionados (nested)
    patient_name: Optional[str] = None
    doctor_name: Optional[str] = None

    class Config:
        from_attributes = True


class MedicalRecordListResponse(BaseModel):
    """Schema para lista de historias clínicas"""
    medical_records: list[MedicalRecordResponse]
    total: int

File: frontend/src/types/medicalRecord.ts
/**
 * Tipos para Medical Records (Historias Clínicas)
 */

export interface MedicalRecord {
  id: number;
  consultorio_id: number;
  patient_id: number;
  appointment_id?: number;
  doctor_id: number;
  consultation_date: string; // ISO 8601 datetime
  chief_complaint?: string;
  symptoms?: string;
  diagnosis: string;
  treatment?: string;
  notes?: string;
  // Signos vitales
  blood_pressure?: string;
  heart_rate?: number;
  temperature?: number;
  weight?: number;
  height?: number;
  created_at: string;
  updated_at: string;
  // Datos anidados
  patient_name?: string;
  doctor_name?: string;
}

export interface MedicalRecordCreate {
  patient_id: number;
  appointment_id?: number;
  consultation_date: string;
  chief_complaint?: string;
  symptoms?: string;
  diagnosis: string;
  treatment?: string;
  notes?: string;
  blood_pressure?: string;
  heart_rate?: number;
  temperature?: number;
  weight?: number;
  height?: number;
}

export interface MedicalRecordUpdate {
  patient_id?: number;
  appointment_id?: number;
  consultation_date?: string;
  chief_complaint?: string;
  symptoms?: string;
  diagnosis?: string;
  treatment?: string;
  notes?: string;
  blood_pressure?: string;
  heart_rate?: number;
  temperature?: number;
  weight?: number;
  height?: number;
}

export interface MedicalRecordFilters {
  patient_id?: number;
  doctor_id?: number;
  start_date?: string;
  end_date?: string;
  skip?: number;
  limit?: number;
}

export interface MedicalRecordListResponse {
  medical_records: MedicalRecord[];
  total: number;
}
